ddtrace/llmobs/_llmobs.py

```diff
@@ -583,8 +584,88 @@ def create_dataset(cls, name: str, description: str, records: List[DatasetRecord
         ds = cls._instance._dne_client.dataset_create(name, description)
         for r in records:
             ds.append(r)
-        if len(records) > 0:
-            ds.push()
+        return ds
+
+    @classmethod
+    def create_dataset_from_csv(
+        cls,
+        csv_path: str,
+        dataset_name: str,
+        input_data_columns: List[str],
+        expected_output_columns: List[str],
+        metadata_columns: List[str] = [],
+        csv_delimiter: str = ",",
+        description="",
+    ) -> Dataset:
+        ds = cls._instance._dne_client.dataset_create(dataset_name, description)
+
+        # Store the original field size limit to restore it later
+        original_field_size_limit = csv.field_size_limit()
+
+        csv.field_size_limit(10 * 1024 * 1024)  # 10mb
+
+        try:
+            # First check if the file exists and is not empty before parsing
+            with open(csv_path, mode="r") as csvfile:
+                content = csvfile.readline().strip()
+                if not content:
+                    raise ValueError("CSV file appears to be empty or header is missing.")
+
+            with open(csv_path, mode="r") as csvfile:
+                try:
+                    rows = csv.DictReader(csvfile, delimiter=csv_delimiter)
+
+                    # Check header presence before trying to read rows
+                    if rows.fieldnames is None:
+                        # Treat files with no header at all as effectively empty
+                        raise ValueError("CSV file appears to be empty or header is missing.")
+
+                    header_columns = rows.fieldnames
+                    missing_input_columns = [col for col in input_data_columns if col not in header_columns]
+                    missing_output_columns = [col for col in expected_output_columns if col not in header_columns]
+
+                    if missing_input_columns:
+                        raise ValueError(f"Input columns not found in CSV header: {missing_input_columns}")
+                    if missing_output_columns:
```

Comment:
Can we not just do this?
```python
if any(col not in header_columns for col in input_data_cols):

if any(col not in header_columns for col in expected_output_columns):
```

Reply:
as someone that just came from working on java this feels illegal but sure we can do this, seems neat